### PR TITLE
[TRACKING FIX] CTA Carousel: branch link tracking param update

### PR DESCRIPTION
**Pull request by qiyundai**
Add q param for branch link tracking on the CTA Carousel
+ a small spacing tweak for toggle bar scrolling

Test URLs:
- Before: https://main--express--adobecom.hlx.page/express/
- After: https://homepage-branch-tracking--express--adobecom.hlx.page/express/

### CODE REVIEW

**Review by JingleH — APPROVED** [review]

File: express/blocks/cta-carousel/cta-carousel.js [hunk 1]
```diff
@@ -165,8 +166,15 @@ export async function decorateCards(block, payload) {
...
         if (a.href && a.href.match('adobesparkpost.app.link')) {
           const btnUrl = new URL(a.href);
-          btnUrl.searchParams.set('search', cta.text);
-          a.href = decodeURIComponent(btnUrl.toString());
+          if (placeholders['search-branch-links']
+            && placeholders['search-branch-links']
+              .replace(/\s/g, '')
+              .split(',')
+              .includes(`${btnUrl.origin}${btnUrl.pathname}`)) {
+            btnUrl.searchParams.set('search', cta.text);
```
Comment (toka1300) [inline express/blocks/cta-carousel/cta-carousel.js:174]: 'search' and 'q' params are set to the same?

File: express/blocks/cta-carousel/cta-carousel.js [hunk 2]
```diff
@@ -165,8 +166,15 @@ export async function decorateCards(block, payload) {
       cta.ctaLinks.forEach((a) => {
         if (a.href && a.href.match('adobesparkpost.app.link')) {
           const btnUrl = new URL(a.href);
-          btnUrl.searchParams.set('search', cta.text);
-          a.href = decodeURIComponent(btnUrl.toString());
+          if (placeholders['search-branch-links']
+            && placeholders['search-branch-links']
```
[the remark this answers is not in the document]
Comment (qiyundai) [reply express/blocks/cta-carousel/cta-carousel.js:170]: It's connected to the following lines 171 - 173. But indeed I can probably reduce this to a 1-liner.

File: express/blocks/cta-carousel/cta-carousel.js (hunk 1 above)
Comment (qiyundai) [reply express/blocks/cta-carousel/cta-carousel.js:174]: Yes. Backward compatibility.

**Comment by JingleH** [comment]
@qiyundai we can merge once the conflict is resolved and that conversation with @toka1300 is resolved!

**Comment by qiyundai** [comment]
> @qiyundai we can merge once the conflict is resolved and that conversation with @toka1300 is resolved!

Yup! Just did!

**Review by JingleH — APPROVED** [review]

**Pull request merged**